rubicon/objc/runtime.py

```diff
@@ -811,7 +811,10 @@ def __call__(self, receiver, *args, convert_args=True, convert_result=True):
                     arg = arg.value
 
                 if issubclass(argtype, objc_block):
-                    if isinstance(arg, Block):
+                    if arg is None:
+                        # allow for 'nil' block args, which some objc methods accept
+                        arg = ns_from_py(arg)
+                    elif isinstance(arg, Block):
                         arg = arg.block
                     else:
                         arg = Block(arg).block
```

Comment:
I have a concern about this line here -- is it possible that the Block() temporary created here gets destroyed before obj-c has a chance to 'claim' it?  In my testing it works ok -- but I am not sure why and/or if it's guaranteed to always work with this temporary in place.  

I could always research more deeply the guarantees Python offers about how long a temporary stays around -- but the reference count on that temporary Block() instance drops to 0 pretty quickly.. so it makes me nervous.

Is this line 100% ok?

Reply:
I'm probably missing something, but right now I don't see why this code would work at all - `Block` objects don't have a `block` attribute, only `ObjCBlockInstance`s do. And I think you're right, there's nothing stopping the block from being collected before its pointer is passed to a method. To prevent this, I think the pointer object would need to keep a reference to the block object - or maybe the block object should simply have an `_as_parameter_` attribute, so it can be passed into C functions as-is.

Reply:
Oh man, thanks for responding.

I can attest to it magically working. I have tested that codepath by passing naked Python callables to obj-c methods with all sorts of semantics from immediate to asynchronous -- and they work -- the copy_helper is invoked pretty much right away without crashing despite everything -- I guess we got lucky -- I'm surprised Python allows for objects with 0 refcount, marked for deletion, to be resurrected later!  Go Python! :)

Indeed the Block class does have the .block attribute, akin to the .ptr attribute for ObjCInstance, which is basically the raw pointer. It's assigned in the very last line of the constructor: 

``self.block = cast(byref(self.literal), objc_block)``


Anyway -- I'll go ahead and update this PR to mitigate the risk of crash created here by doing either what I propose (a list), or I can investigate this _as_parameter_ mechanism you speak of (not familiar with it but hey, an opportunity to learn!)

Thanks for responding quickly!